power_grid_model_c/power_grid_model/include/power_grid_model/main_model_impl.hpp

```diff
@@ -634,10 +643,16 @@ class MainModelImpl<ExtraRetrievableTypes<ExtraRetrievableType...>, ComponentLis
             };
             auto model = copy_model_functor(start);
 
-            SequenceIdx cacheable_scenario_sequence = SequenceIdx{};
-            auto const& scenario_sequence = is_independent ? all_scenarios_sequence : cacheable_scenario_sequence;
-            auto [setup, winddown] = scenario_update_restore(model, update_data, scenario_sequence,
-                                                             cacheable_scenario_sequence, is_independent, infos);
+            SequenceIdx current_scenario_sequence_cache = SequenceIdx{};
+            std::array<std::reference_wrapper<std::vector<Idx2D> const>, n_types> current_scenario_sequence =
+                run_functor_with_all_types_return_array(
+                    [&is_independent, &all_scenarios_sequence, &current_scenario_sequence_cache]<typename CT>() {
+                        constexpr auto comp_idx = index_of_component<CT>;
+                        return is_independent[comp_idx] ? std::cref(all_scenarios_sequence[comp_idx])
+                                                        : std::cref(current_scenario_sequence_cache[comp_idx]);
```

Comment:
`reference_wrapper` is needed here because there is no default constructor for a raw reference

Reply:
hmmm. interesting approach... you may be right if we pass the sequence of this component type by view...

EDIT: I don't think this will work because the size of the vector may differ between scenarios. pass-by-value semantics (`std::span`) is not updated, but `std::reference_wrapper` is.

Reply:
Indeed this doesn't work when the size of the vector changes. Then the reference wrapper is the right approach.